fenix/app/src/main/java/org/mozilla/fenix/translations/TranslationOptionsDialog.kt

```diff
@@ -80,13 +119,41 @@ fun TranslationOptionsDialog(
 }
 
 @Composable
-private fun TranslationOptions(translationSwitchItem: TranslationSwitchItem) {
+private fun EnabledTranslationOptionsItems(
+    translationOptionsListState: SnapshotStateList<TranslationSwitchItem>,
+) {
+    val itemCheckedWithHighestImportance =
+        translationOptionsListState.sortedByDescending { listItem -> listItem.importance }
+            .firstOrNull { it.isChecked }
+
+    // If no items are checked, then all the switches should be enabled.
+    if (itemCheckedWithHighestImportance == null || itemCheckedWithHighestImportance.importance == 0) {
```

Comment:
I'm still thinking about how we will connect all of these calls and manage state on this screen. I think this will set us up to manage the view state in a nice way. 

What are your thoughts on organizing those calls?

Reply:
(Conversation happened at [other comment](https://github.com/mozilla-mobile/firefox-android/pull/5171#discussion_r1454148224))